source/common/http/conn_manager_impl.cc

```diff
@@ -404,6 +423,15 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(HeaderMapPtr&& headers,
   // Set the trusted address for the connection by taking the last address in XFF.
   downstream_address_ = Utility::getLastAddressFromXFF(*request_headers_);
   decodeHeaders(nullptr, *request_headers_, end_stream);
+
+  Tracing::Decision tracing_decision =
```

Comment:
It will be tricky to do filtering in finish span, as we already propagated context to the upstream service and upstream could propagate this hierarchy further (when parent/child will be in place), although this is not applicable for HC specifically (there is no further propagation of HC requests).

But i could do filtering before creating span, actually i think HC is the only thing bothers me, so probably i can have some easy way to detect HC.